### PR TITLE
add wallet dir creation for bitcoin core 

### PR DESCRIPTION
**Pull request by Kukks**

### CODE REVIEW

**Comment by NicolasDorier** [comment]
Would be good, but you need to update https://github.com/btcpayserver/dockerfile-deps instead :D

File: core/0.18.0/docker-entrypoint.sh [hunk 1]
```diff
@@ -15,6 +15,14 @@ if [[ "$1" == "bitcoin-cli" || "$1" == "bitcoin-tx" || "$1" == "bitcoind" || "$1
 		CONFIG_PREFIX=$'mainnet=1\n[main]'
 	fi
 
+	if [[ $BITCOIN_WALLETDIR ]]; then
+		NL=$'\n'
+		WALLETDIR="$BITCOIN_WALLETDIR/${BITCOIN_NETWORK}"
+		mkdir -p $WALLETDIR
```
Comment (NicolasDorier) [inline core/0.18.0/docker-entrypoint.sh]: Need quotes `"`

**Comment by Kukks** [comment]
If you think it is fine, i will open pr to https://github.com/btcpayserver/dockerfile-deps/tree/master/Bitcoin/0.18.0 too

File: core/0.18.0/Dockerfile [hunk 2]
```diff
@@ -13,7 +13,7 @@ ENV BITCOIN_URL https://bitcoincore.org/bin/bitcoin-core-0.18.0/
 ENV BITCOIN_SHA256 5146ac5310133fbb01439666131588006543ab5364435b748ddfc95a8cb8d63f
 ENV BITCOIN_ASC_URL https://bitcoincore.org/bin/bitcoin-core-0.18.0/SHA256SUMS.asc
 ENV BITCOIN_PGP_KEY 01EA5486DE18A882D4C2684590C8019E36C2E964
-
+ENV BITCOIN_NETWORK mainnet
```
Comment (NicolasDorier) [inline core/0.18.0/Dockerfile]: Remove this

File: core/0.18.0/docker-entrypoint.sh [hunk 3]
```diff
@@ -15,6 +15,14 @@ if [[ "$1" == "bitcoin-cli" || "$1" == "bitcoin-tx" || "$1" == "bitcoind" || "$1
 		CONFIG_PREFIX=$'mainnet=1\n[main]'
 	fi
 
+	if [[ "$BITCOIN_WALLETDIR" ]]; then
+		NL=$'\n'
+		WALLETDIR="$BITCOIN_WALLETDIR/${BITCOIN_NETWORK}"
```
Comment (NicolasDorier) [inline core/0.18.0/docker-entrypoint.sh]: Use `${BITCOIN_NETWORK:-mainnet}`

File: core/0.18.0/docker-entrypoint.sh [hunk 4]
```diff
@@ -15,6 +15,14 @@ if [[ "$1" == "bitcoin-cli" || "$1" == "bitcoin-tx" || "$1" == "bitcoind" || "$1
 		CONFIG_PREFIX=$'mainnet=1\n[main]'
 	fi
 
+	if [[ "$BITCOIN_WALLETDIR" ]]; then
```
Comment (NicolasDorier) [inline core/0.18.0/docker-entrypoint.sh]: add `[[ "$BITCOIN_WALLETDIR" ]] && [[ "$BITCOIN_NETWORK" ]]`

File: core/0.18.0/docker-entrypoint.sh [hunk 5]
```diff
@@ -15,6 +15,14 @@ if [[ "$1" == "bitcoin-cli" || "$1" == "bitcoin-tx" || "$1" == "bitcoind" || "$1
 		CONFIG_PREFIX=$'mainnet=1\n[main]'
 	fi
```
Comment (NicolasDorier) [inline core/0.18.0/docker-entrypoint.sh:16]: We should use `elif` instead with `else` which reset `BITCOIN_NETWORK` to empty string if it does not match anything.

File: example/docker-compose.yml [hunk 6]
```diff
@@ -1,20 +1,20 @@
...
   bitcoind:
-    image: NicolasDorier/bitcoin
+    image: nicolasdorier/docker-bitcoin:0.18.0
     # specify additional command line arguments here
     command: bitcoind
     healthcheck:
       test: ["CMD", "/entrypoint.sh", "bitcoin-cli", "getinfo"]
     ports:
       # change to 18333 if running testnet
-      - "8333:8333"
```
Comment (NicolasDorier) [inline example/docker-compose.yml]: should be 8333

**Pull request merged**